tests/macrodata/test_pynsee_macrodata.py

```diff
@@ -116,7 +117,11 @@ def test_get_date(self):
             "001580394",
         )
         assert isinstance(data, pd.DataFrame)
-        assert _get_date(freq="TEST", time_period=3) == 3
+        df = pd.DataFrame([[3, "blah"]], columns=["TIME_PERIOD", "FREQ"])
+        _set_date(df)
+        assert (
+            df["DATE"].values == df["TIME_PERIOD"].values == np.array(3)
+        ).all()
```

Comment:
if you have time, could you add actual tests for the different Q/S/B variants?

Reply:
I just checked: in fact, `get_series("001694056", "001691912", "001580062", "001688370", "010565692", "001580394")` returns a dataframe with all variations on `"TIME_PERIOD"`. That's the reason there was only a dummy test on `_set_date`.